src/instance.rs

```diff
@@ -172,15 +172,15 @@ impl<'py, T> Py2<'py, T> {
         self.into_non_null().as_ptr()
     }
 
-    /// Internal helper to convert e.g. &'a &'py PyDict to &'a Py2<'py, PyDict> for
+    /// Internal helper to convert e.g. &'a &'py PyDict to &'a Bound<'py, PyDict> for
     /// backwards-compatibility during migration to removal of pool.
     #[doc(hidden)] // public and doc(hidden) to use in examples and tests for now
```

Comment:
Should this stay hidden if `Bound` becomes part of the public API? Shouldn't users also be able to switch between GIL references and `Bound` smart pointers?

Reply:
Are you willing to accept this to stay as-is just for the moment, and then in the follow up adding `PyNativeType::as_bound` maybe we can delete this completely?

Reply:
Alright, let's consider this together with `as_bound`.